pkgs/development/python-modules/pydeck/default.nix

```diff
@@ -0,0 +1,25 @@
+{ lib, buildPythonApplication, fetchPypi, pkgs }:
+
+buildPythonApplication rec {
+  pname = "pydeck";
+  version = "0.7.0";
+
+  src = pkgs.python3Packages.fetchPypi {
+    inherit pname version;
+    sha256 = "1zi0gqzd0byj16ja74m2dm99a1hmrlhk26y0x7am07vb1d8lvvsy";
+  };
+
+  doCheck = false; #can't find pytest
```

Comment:
What did you try?

Reply:
with `checkInputs = [ pytest ];` i still get same error with just as with `doCheck = true;`
tried adding `pytest` to `buildInputs`,`propagatedBuildInputs` still same

```
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
setuptoolsCheckPhase
Executing setuptoolsCheckPhase
/nix/store/1rp4sajaq2gb2cf1s5iqw23l2bj30vlx-python3.9-setuptools-57.2.0/lib/python3.9/site-packages/setuptools/dist.py:697: UserWarning: Usage of dash-separated 'description-file' will not be supported in future versions. Please use the underscore name 'description_file' instead
  warnings.warn(
usage: nix_run_setup [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: nix_run_setup --help [cmd1 cmd2 ...]
   or: nix_run_setup --help-commands
   or: nix_run_setup cmd --help

error: invalid command 'pytest'
builder for '/nix/store/3s3x33nygwz3qjhx7f7j9pza8q03jcmx-python3.9-pydeck-0.7.0.drv' failed with exit code 1
error: build of '/nix/store/3s3x33nygwz3qjhx7f7j9pza8q03jcmx-python3.9-pydeck-0.7.0.drv' failed
```
i saw `preCheck` so tried `preCheck =  [ pytest ];`
```
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
setuptoolsCheckPhase
Executing setuptoolsCheckPhase
/nix/store/9jsp95c5r4pl1pdqzy9z405wj39czpwa-stdenv-linux/setup: line 87: /nix/store/c7n589hij1068fv58ycb72ijrr7nmrf0-python3.9-pytest-6.2.5: Is a directory
builder for '/nix/store/j93hs2dykcnmfpa9as0ykafw7464qswr-python3.9-pydeck-0.7.0.drv' failed with exit code 126
error: build of '/nix/store/j93hs2dykcnmfpa9as0ykafw7464qswr-python3.9-pydeck-0.7.0.drv' failed
```

Reply:
`checkInputs = [ pytestCheckHook ]` ?

Reply:
> `checkInputs = [ pytestCheckHook ]` ?

```
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
pytestCheckPhase
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.9.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: /build/pydeck-0.7.0, configfile: pytest.ini, testpaths: tests/
collected 19 items / 3 errors / 16 selected                                    

==================================== ERRORS ====================================
__________________ ERROR collecting tests/test_data_utils.py ___________________
ImportError while importing test module '/build/pydeck-0.7.0/tests/test_data_utils.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_data_utils.py:1: in <module>
    import pandas as pd
E   ModuleNotFoundError: No module named 'pandas'
_________________ ERROR collecting tests/bindings/test_deck.py _________________
ImportError while importing test module '/build/pydeck-0.7.0/tests/bindings/test_deck.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/bindings/test_deck.py:8: in <module>
    from . import pydeck_examples
tests/bindings/pydeck_examples/__init__.py:8: in <module>
    from .stacked import create_stacked_test_object  # noqa
tests/bindings/pydeck_examples/stacked.py:1: in <module>
    import pandas as pd
E   ModuleNotFoundError: No module named 'pandas'
------------------------------- Captured stderr --------------------------------
Exception ignored in: <_io.FileIO name='/build/pydeck-0.7.0/tests/./fixtures/minimal.json' mode='rb' closefd=True>
Traceback (most recent call last):
  File "/build/pydeck-0.7.0/tests/fixtures.py", line 14, in <module>
    fixture_text = open(fname).read()
ResourceWarning: unclosed file <_io.TextIOWrapper name='/build/pydeck-0.7.0/tests/./fixtures/minimal.json' mode='r' encoding='UTF-8'>
Exception ignored in: <_io.FileIO name='/build/pydeck-0.7.0/tests/./fixtures/stacked.json' mode='rb' closefd=True>
Traceback (most recent call last):
  File "/build/pydeck-0.7.0/tests/fixtures.py", line 14, in <module>
    fixture_text = open(fname).read()
ResourceWarning: unclosed file <_io.TextIOWrapper name='/build/pydeck-0.7.0/tests/./fixtures/stacked.json' mode='r' encoding='UTF-8'>
Exception ignored in: <_io.FileIO name='/build/pydeck-0.7.0/tests/./fixtures/multilayers.json' mode='rb' closefd=True>
Traceback (most recent call last):
  File "/build/pydeck-0.7.0/tests/fixtures.py", line 14, in <module>
    fixture_text = open(fname).read()
ResourceWarning: unclosed file <_io.TextIOWrapper name='/build/pydeck-0.7.0/tests/./fixtures/multilayers.json' mode='r' encoding='UTF-8'>
Exception ignored in: <_io.FileIO name='/build/pydeck-0.7.0/tests/./fixtures/local-image.json' mode='rb' closefd=True>
Traceback (most recent call last):
  File "/build/pydeck-0.7.0/tests/fixtures.py", line 14, in <module>
    fixture_text = open(fname).read()
ResourceWarning: unclosed file <_io.TextIOWrapper name='/build/pydeck-0.7.0/tests/./fixtures/local-image.json' mode='r' encoding='UTF-8'>
Exception ignored in: <_io.FileIO name='/build/pydeck-0.7.0/tests/./fixtures/scatterplot.json' mode='rb' closefd=True>
Traceback (most recent call last):
  File "/build/pydeck-0.7.0/tests/fixtures.py", line 14, in <module>
    fixture_text = open(fname).read()
ResourceWarning: unclosed file <_io.TextIOWrapper name='/build/pydeck-0.7.0/tests/./fixtures/scatterplot.json' mode='r' encoding='UTF-8'>
Exception ignored in: <_io.FileIO name='/build/pydeck-0.7.0/tests/./fixtures/geojson-layer-with-gmaps.json' mode='rb' closefd=True>
Traceback (most recent call last):
  File "/build/pydeck-0.7.0/tests/fixtures.py", line 14, in <module>
    fixture_text = open(fname).read()
ResourceWarning: unclosed file <_io.TextIOWrapper name='/build/pydeck-0.7.0/tests/./fixtures/geojson-layer-with-gmaps.json' mode='r' encoding='UTF-8'>
Exception ignored in: <_io.FileIO name='/build/pydeck-0.7.0/tests/./fixtures/geojson-layer.json' mode='rb' closefd=True>
Traceback (most recent call last):
  File "/build/pydeck-0.7.0/tests/fixtures.py", line 14, in <module>
    fixture_text = open(fname).read()
ResourceWarning: unclosed file <_io.TextIOWrapper name='/build/pydeck-0.7.0/tests/./fixtures/geojson-layer.json' mode='r' encoding='UTF-8'>
Exception ignored in: <_io.FileIO name='/build/pydeck-0.7.0/tests/./fixtures/heatmap-layer.json' mode='rb' closefd=True>
Traceback (most recent call last):
  File "/build/pydeck-0.7.0/tests/fixtures.py", line 14, in <module>
    fixture_text = open(fname).read()
ResourceWarning: unclosed file <_io.TextIOWrapper name='/build/pydeck-0.7.0/tests/./fixtures/heatmap-layer.json' mode='r' encoding='UTF-8'>
Exception ignored in: <_io.FileIO name='/build/pydeck-0.7.0/tests/./fixtures/heatmap-layer-function.json' mode='rb' closefd=True>
Traceback (most recent call last):
  File "/build/pydeck-0.7.0/tests/fixtures.py", line 14, in <module>
    fixture_text = open(fname).read()
ResourceWarning: unclosed file <_io.TextIOWrapper name='/build/pydeck-0.7.0/tests/./fixtures/heatmap-layer-function.json' mode='r' encoding='UTF-8'>
________________ ERROR collecting tests/bindings/test_layer.py _________________
ImportError while importing test module '/build/pydeck-0.7.0/tests/bindings/test_layer.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/bindings/test_layer.py:1: in <module>
    import pandas as pd
E   ModuleNotFoundError: No module named 'pandas'
=========================== short test summary info ============================
ERROR tests/test_data_utils.py
ERROR tests/bindings/test_deck.py
ERROR tests/bindings/test_layer.py
!!!!!!!!!!!!!!!!!!! Interrupted: 3 errors during collection !!!!!!!!!!!!!!!!!!!!
============================== 3 errors in 0.43s ===============================
builder for '/nix/store/bdj473p2dfnn7k1mgsnrhvjlz2hr06q7-python3.9-pydeck-0.7.0.drv' failed with exit code 2
error: build of '/nix/store/bdj473p2dfnn7k1mgsnrhvjlz2hr06q7-python3.9-pydeck-0.7.0.drv' failed
```

Reply:
> `checkInputs = [ pytestCheckHook ]` ?
added `pandas` and `jupyter` cuz this need it.

now:

```
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.9.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: /build/pydeck-0.7.0, configfile: pytest.ini, testpaths: tests/
collecting ... 
----------------------------- live log collection ------------------------------
2021-09-26 21:04:48 INFO NumExpr defaulting to 8 threads.
collected 31 items                                                             

tests/test_data_utils.py::test_euclidean PASSED                          [  3%]
tests/test_data_utils.py::test_bbox_to_zoom_level PASSED                 [  6%]
tests/test_data_utils.py::test_geometric_mean PASSED                     [  9%]
tests/test_data_utils.py::test_k_nearest_neighbors PASSED                [ 12%]
tests/test_data_utils.py::test_get_n_pct PASSED                          [ 16%]
tests/test_data_utils.py::test_get_bbox PASSED                           [ 19%]
tests/test_data_utils.py::test_is_pandas_df PASSED                       [ 22%]
tests/test_data_utils.py::test_compute_view PASSED                       [ 25%]
tests/test_deckgl_widget.py::test_example_creation_blank PASSED          [ 29%]
tests/test_nbextension_path.py::test_nbextension_path PASSED             [ 32%]
tests/test_settings.py::test_settings_is_imported PASSED                 [ 35%]
tests/bindings/test_deck.py::test_deck_layer_args PASSED                 [ 38%]
tests/bindings/test_deck.py::test_json_output PASSED                     [ 41%]
tests/bindings/test_deck.py::test_update PASSED                          [ 45%]
tests/bindings/test_image.py::test_url_case PASSED                       [ 48%]
tests/bindings/test_image.py::test_local_file_case PASSED                [ 51%]
tests/bindings/test_layer.py::test_constructor_binary_transport PASSED   [ 54%]
tests/bindings/test_string.py::test_basic_case PASSED                    [ 58%]
tests/bindings/test_string.py::test_quotes PASSED                        [ 61%]
tests/bindings/test_view.py::test_view_constructor PASSED                [ 64%]
tests/bindings/data_utils/test_binary_transfer.py::test_serialize_columns_none_input PASSED [ 67%]
tests/bindings/data_utils/test_binary_transfer.py::test_serialize_columns PASSED [ 70%]
tests/browser/test_examples.py::test_notebooks SKIPPED (No golden im...) [ 74%]
tests/browser/test_html_renderer.py::test_standalone_rendering SKIPPED   [ 77%]
tests/browser/test_html_renderer.py::test_notebook_iframe_rendering SKIPPED [ 80%]
tests/io/test_html.py::test_rendering_is_not_broken PASSED               [ 83%]
tests/io/test_html.py::test_display_html PASSED                          [ 87%]
tests/io/test_html.py::test_cdn_picker PASSED                            [ 90%]
tests/io/test_html.py::test_iframe_with_srcdoc PASSED                    [ 93%]
tests/io/test_html.py::test_in_jupyter PASSED                            [ 96%]
tests/notebooks/test_nbconvert.py::test_nbconvert FAILED                 [100%]

=================================== FAILURES ===================================
________________________________ test_nbconvert ________________________________

    def test_nbconvert():
        for fname in glob.glob(nb_glob):
            # NOTE Massive data sets notebook takes too long to render, skipping for now
            # NOTE skipping 03 (which contains live singapore taxi data) until I can provide a static version of that data
            if "04" in fname or "06" in fname or "03" in fname:
                continue
            print(fname)
>           assert nbconvert(fname)

tests/notebooks/test_nbconvert.py:18:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/notebooks/notebook_utils.py:15: in nbconvert
    status = subprocess.check_call(cmd, shell=True)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

popenargs = ("jupyter nbconvert --to=html --ExecutePreprocessor.timeout=600 --ExecutePreprocessor.kernel_name='python3' --ExecuteP...rue --execute '/build/pydeck-0.7.0/tests/notebooks/../../examples/05 - Interacting with other Jupyter widgets.ipynb'",)
kwargs = {'shell': True}, retcode = 1
cmd = "jupyter nbconvert --to=html --ExecutePreprocessor.timeout=600 --ExecutePreprocessor.kernel_name='python3' --ExecutePr...=True --execute '/build/pydeck-0.7.0/tests/notebooks/../../examples/05 - Interacting with other Jupyter widgets.ipynb'"

    def check_call(*popenargs, **kwargs):
        """Run command with arguments.  Wait for command to complete.  If
        the exit code was zero then return, otherwise raise
        CalledProcessError.  The CalledProcessError object will have the
        return code in the returncode attribute.
    
        The arguments are the same as for the call function.  Example:
    
        check_call(["ls", "-l"])
        """
        retcode = call(*popenargs, **kwargs)
        if retcode:
            cmd = kwargs.get("args")
            if cmd is None:
                cmd = popenargs[0]
>           raise CalledProcessError(retcode, cmd)
E           subprocess.CalledProcessError: Command 'jupyter nbconvert --to=html --ExecutePreprocessor.timeout=600 --ExecutePreprocessor.kernel_name='python3' --ExecutePreprocessor.store_widget_state=True --execute '/build/pydeck-0.7.0/tests/notebooks/../../examples/05 - Interacting with other Jupyter widgets.ipynb'' returned non-zero exit status 1.

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/subprocess.py:373: CalledProcessError
----------------------------- Captured stdout call -----------------------------
/build/pydeck-0.7.0/tests/notebooks/../../examples/05 - Interacting with other Jupyter widgets.ipynb
----------------------------- Captured stderr call -----------------------------
[NbConvertApp] Converting notebook /build/pydeck-0.7.0/tests/notebooks/../../examples/05 - Interacting with other Jupyter widgets.ipynb to html
/nix/store/rv0j6l69jp8p8fypfxakbw8g6cglq6n7-python3.9-ipython-7.27.0/lib/python3.9/site-packages/IPython/paths.py:66: UserWarning: IPython parent '/homeless-shelter' is not a writable location, using a temp directory.
  warn("IPython parent '{0}' is not a writable location,"
/nix/store/rv0j6l69jp8p8fypfxakbw8g6cglq6n7-python3.9-ipython-7.27.0/lib/python3.9/site-packages/IPython/paths.py:66: UserWarning: IPython parent '/homeless-shelter' is not a writable location, using a temp directory.
  warn("IPython parent '{0}' is not a writable location,"
Assertion failed: pfd.revents & POLLIN (/build/source/src/signaler.cpp:265)
Traceback (most recent call last):
  File "/nix/store/nh8v866dfdwyzfxq8jhca17n8k19m7h6-python3.9-nbconvert-6.1.0/bin/.jupyter-nbconvert-wrapped", line 9, in <module>
    sys.exit(main())
  File "/nix/store/amdr323x4x6vhiq605gx133vvljxqll3-python3.9-jupyter_core-4.7.1/lib/python3.9/site-packages/jupyter_core/application.py", line 254, in launch_instance
    return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs)
  File "/nix/store/r2fsrxr723n70a4ry006kbdmbpxhvb4x-python3.9-traitlets-5.1.0/lib/python3.9/site-packages/traitlets/config/application.py", line 846, in launch_instance
    app.start()
  File "/nix/store/nh8v866dfdwyzfxq8jhca17n8k19m7h6-python3.9-nbconvert-6.1.0/lib/python3.9/site-packages/nbconvert/nbconvertapp.py", line 346, in start
    self.convert_notebooks()
  File "/nix/store/nh8v866dfdwyzfxq8jhca17n8k19m7h6-python3.9-nbconvert-6.1.0/lib/python3.9/site-packages/nbconvert/nbconvertapp.py", line 518, in convert_notebooks
    self.convert_single_notebook(notebook_filename)
  File "/nix/store/nh8v866dfdwyzfxq8jhca17n8k19m7h6-python3.9-nbconvert-6.1.0/lib/python3.9/site-packages/nbconvert/nbconvertapp.py", line 483, in convert_single_notebook
    output, resources = self.export_single_notebook(notebook_filename, resources, input_buffer=input_buffer)
  File "/nix/store/nh8v866dfdwyzfxq8jhca17n8k19m7h6-python3.9-nbconvert-6.1.0/lib/python3.9/site-packages/nbconvert/nbconvertapp.py", line 412, in export_single_notebook
    output, resources = self.exporter.from_filename(notebook_filename, resources=resources)
  File "/nix/store/nh8v866dfdwyzfxq8jhca17n8k19m7h6-python3.9-nbconvert-6.1.0/lib/python3.9/site-packages/nbconvert/exporters/exporter.py", line 181, in from_filename
    return self.from_file(f, resources=resources, **kw)
  File "/nix/store/nh8v866dfdwyzfxq8jhca17n8k19m7h6-python3.9-nbconvert-6.1.0/lib/python3.9/site-packages/nbconvert/exporters/exporter.py", line 199, in from_file
    return self.from_notebook_node(nbformat.read(file_stream, as_version=4), resources=resources, **kw)
  File "/nix/store/nh8v866dfdwyzfxq8jhca17n8k19m7h6-python3.9-nbconvert-6.1.0/lib/python3.9/site-packages/nbconvert/exporters/html.py", line 129, in from_notebook_node
    return super().from_notebook_node(nb, resources, **kw)
  File "/nix/store/nh8v866dfdwyzfxq8jhca17n8k19m7h6-python3.9-nbconvert-6.1.0/lib/python3.9/site-packages/nbconvert/exporters/templateexporter.py", line 374, in from_notebook_node
    nb_copy, resources = super().from_notebook_node(nb, resources, **kw)
  File "/nix/store/nh8v866dfdwyzfxq8jhca17n8k19m7h6-python3.9-nbconvert-6.1.0/lib/python3.9/site-packages/nbconvert/exporters/exporter.py", line 143, in from_notebook_node
    nb_copy, resources = self._preprocess(nb_copy, resources)
  File "/nix/store/nh8v866dfdwyzfxq8jhca17n8k19m7h6-python3.9-nbconvert-6.1.0/lib/python3.9/site-packages/nbconvert/exporters/exporter.py", line 318, in _preprocess
    nbc, resc = preprocessor(nbc, resc)
  File "/nix/store/nh8v866dfdwyzfxq8jhca17n8k19m7h6-python3.9-nbconvert-6.1.0/lib/python3.9/site-packages/nbconvert/preprocessors/base.py", line 47, in __call__
    return self.preprocess(nb, resources)
  File "/nix/store/nh8v866dfdwyzfxq8jhca17n8k19m7h6-python3.9-nbconvert-6.1.0/lib/python3.9/site-packages/nbconvert/preprocessors/execute.py", line 84, in preprocess
    self.preprocess_cell(cell, resources, index)
  File "/nix/store/nh8v866dfdwyzfxq8jhca17n8k19m7h6-python3.9-nbconvert-6.1.0/lib/python3.9/site-packages/nbconvert/preprocessors/execute.py", line 105, in preprocess_cell
    cell = self.execute_cell(cell, index, store_history=True)
  File "/nix/store/m0hrd8dr9x14cajfrwnmhsixql14msir-python3.9-nbclient-0.5.4/lib/python3.9/site-packages/nbclient/util.py", line 78, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/nix/store/m0hrd8dr9x14cajfrwnmhsixql14msir-python3.9-nbclient-0.5.4/lib/python3.9/site-packages/nbclient/util.py", line 57, in just_run
    return loop.run_until_complete(coro)
  File "/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/nix/store/m0hrd8dr9x14cajfrwnmhsixql14msir-python3.9-nbclient-0.5.4/lib/python3.9/site-packages/nbclient/client.py", line 862, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/nix/store/m0hrd8dr9x14cajfrwnmhsixql14msir-python3.9-nbclient-0.5.4/lib/python3.9/site-packages/nbclient/client.py", line 765, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
LIGHTS_URL = 'https://raw.githubusercontent.com/ajduberstein/lights_at_night/master/chengdu_lights_at_night.csv'
df = pd.read_csv(LIGHTS_URL)
df.head()
------------------

---------------------------------------------------------------------------
gaierror                                  Traceback (most recent call last)
/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/urllib/request.py in do_open(self, http_class, req, **http_conn_args)
   1345             try:
-> 1346                 h.request(req.get_method(), req.selector, req.data, headers,
   1347                           encode_chunked=req.has_header('Transfer-encoding'))

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/http/client.py in request(self, method, url, body, headers, encode_chunked)
   1256         """Send a complete request to the server."""
-> 1257         self._send_request(method, url, body, headers, encode_chunked)
   1258 

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/http/client.py in _send_request(self, method, url, body, headers, encode_chunked)
   1302             body = _encode(body, 'body')
-> 1303         self.endheaders(body, encode_chunked=encode_chunked)
   1304 

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/http/client.py in endheaders(self, message_body, encode_chunked)
   1251             raise CannotSendHeader()
-> 1252         self._send_output(message_body, encode_chunked=encode_chunked)
   1253 

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/http/client.py in _send_output(self, message_body, encode_chunked)
   1011         del self._buffer[:]
-> 1012         self.send(msg)
   1013 

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/http/client.py in send(self, data)
    951             if self.auto_open:
--> 952                 self.connect()
    953             else:

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/http/client.py in connect(self)
   1418 
-> 1419             super().connect()
   1420 

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/http/client.py in connect(self)
    922         """Connect to the host and port specified in __init__."""
--> 923         self.sock = self._create_connection(
    924             (self.host,self.port), self.timeout, self.source_address)

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/socket.py in create_connection(address, timeout, source_address)
    821     err = None
--> 822     for res in getaddrinfo(host, port, 0, SOCK_STREAM):
    823         af, socktype, proto, canonname, sa = res

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/socket.py in getaddrinfo(host, port, family, type, proto, flags)
    952     addrlist = []
--> 953     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
    954         af, socktype, proto, canonname, sa = res

gaierror: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

URLError                                  Traceback (most recent call last)
/build/ipykernel_321/2256898823.py in <module>
      1 LIGHTS_URL = 'https://raw.githubusercontent.com/ajduberstein/lights_at_night/master/chengdu_lights_at_night.csv'
----> 2 df = pd.read_csv(LIGHTS_URL)
      3 df.head()

/nix/store/11lz501j21xwi56b7v2c594s1x7rrgy0-python3.9-pandas-1.3.2/lib/python3.9/site-packages/pandas/util/_decorators.py in wrapper(*args, **kwargs)
    309                     stacklevel=stacklevel,
    310                 )
--> 311             return func(*args, **kwargs)
    312 
    313         return wrapper

/nix/store/11lz501j21xwi56b7v2c594s1x7rrgy0-python3.9-pandas-1.3.2/lib/python3.9/site-packages/pandas/io/parsers/readers.py in read_csv(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, error_bad_lines, warn_bad_lines, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options)
    584     kwds.update(kwds_defaults)
    585 
--> 586     return _read(filepath_or_buffer, kwds)
    587 
    588 

/nix/store/11lz501j21xwi56b7v2c594s1x7rrgy0-python3.9-pandas-1.3.2/lib/python3.9/site-packages/pandas/io/parsers/readers.py in _read(filepath_or_buffer, kwds)
    480 
    481     # Create the parser.
--> 482     parser = TextFileReader(filepath_or_buffer, **kwds)
    483 
    484     if chunksize or iterator:

/nix/store/11lz501j21xwi56b7v2c594s1x7rrgy0-python3.9-pandas-1.3.2/lib/python3.9/site-packages/pandas/io/parsers/readers.py in __init__(self, f, engine, **kwds)
    809             self.options["has_index_names"] = kwds["has_index_names"]
    810 
--> 811         self._engine = self._make_engine(self.engine)
    812 
    813     def close(self):

/nix/store/11lz501j21xwi56b7v2c594s1x7rrgy0-python3.9-pandas-1.3.2/lib/python3.9/site-packages/pandas/io/parsers/readers.py in _make_engine(self, engine)
   1038             )
   1039         # error: Too many arguments for "ParserBase"
-> 1040         return mapping[engine](self.f, **self.options)  # type: ignore[call-arg]
   1041 
   1042     def _failover_to_python(self):

/nix/store/11lz501j21xwi56b7v2c594s1x7rrgy0-python3.9-pandas-1.3.2/lib/python3.9/site-packages/pandas/io/parsers/c_parser_wrapper.py in __init__(self, src, **kwds)
     49 
     50         # open handles
---> 51         self._open_handles(src, kwds)
     52         assert self.handles is not None
     53 

/nix/store/11lz501j21xwi56b7v2c594s1x7rrgy0-python3.9-pandas-1.3.2/lib/python3.9/site-packages/pandas/io/parsers/base_parser.py in _open_handles(self, src, kwds)
    220         Let the readers open IOHandles after they are done with their potential raises.
    221         """
--> 222         self.handles = get_handle(
    223             src,
    224             "r",

/nix/store/11lz501j21xwi56b7v2c594s1x7rrgy0-python3.9-pandas-1.3.2/lib/python3.9/site-packages/pandas/io/common.py in get_handle(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)
    606 
    607     # open URLs
--> 608     ioargs = _get_filepath_or_buffer(
    609         path_or_buf,
    610         encoding=encoding,

/nix/store/11lz501j21xwi56b7v2c594s1x7rrgy0-python3.9-pandas-1.3.2/lib/python3.9/site-packages/pandas/io/common.py in _get_filepath_or_buffer(filepath_or_buffer, encoding, compression, mode, storage_options)
    309         # assuming storage_options is to be interpreted as headers
    310         req_info = urllib.request.Request(filepath_or_buffer, headers=storage_options)
--> 311         with urlopen(req_info) as req:
    312             content_encoding = req.headers.get("Content-Encoding", None)
    313             if content_encoding == "gzip":

/nix/store/11lz501j21xwi56b7v2c594s1x7rrgy0-python3.9-pandas-1.3.2/lib/python3.9/site-packages/pandas/io/common.py in urlopen(*args, **kwargs)
    209     import urllib.request
    210 
--> 211     return urllib.request.urlopen(*args, **kwargs)
    212 
    213 

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/urllib/request.py in urlopen(url, data, timeout, cafile, capath, cadefault, context)
    212     else:
    213         opener = _opener
--> 214     return opener.open(url, data, timeout)
    215 
    216 def install_opener(opener):

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/urllib/request.py in open(self, fullurl, data, timeout)
    515 
    516         sys.audit('urllib.Request', req.full_url, req.data, req.headers, req.get_method())
--> 517         response = self._open(req, data)
    518 
    519         # post-process response

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/urllib/request.py in _open(self, req, data)
    532 
    533         protocol = req.type
--> 534         result = self._call_chain(self.handle_open, protocol, protocol +
    535                                   '_open', req)
    536         if result:

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/urllib/request.py in _call_chain(self, chain, kind, meth_name, *args)
    492         for handler in handlers:
    493             func = getattr(handler, meth_name)
--> 494             result = func(*args)
    495             if result is not None:
    496                 return result

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/urllib/request.py in https_open(self, req)
   1387 
   1388         def https_open(self, req):
-> 1389             return self.do_open(http.client.HTTPSConnection, req,
   1390                 context=self._context, check_hostname=self._check_hostname)
   1391 

/nix/store/dqxic3j7csd4ywn94n4smmnz55p039g3-python3-3.9.6/lib/python3.9/urllib/request.py in do_open(self, http_class, req, **http_conn_args)
   1347                           encode_chunked=req.has_header('Transfer-encoding'))
   1348             except OSError as err: # timeout error
-> 1349                 raise URLError(err)
   1350             r = h.getresponse()
   1351         except:

URLError: <urlopen error [Errno -2] Name or service not known>
URLError: <urlopen error [Errno -2] Name or service not known>

=========================== short test summary info ============================
FAILED tests/notebooks/test_nbconvert.py::test_nbconvert - subprocess.CalledP...
=================== 1 failed, 27 passed, 3 skipped in 11.97s ===================
builder for '/nix/store/cygglx4k33wmy5sx5wfxvdxvsy18dsb6-python3.9-pydeck-0.7.0.drv' failed with exit code 1
error: build of '/nix/store/cygglx4k33wmy5sx5wfxvdxvsy18dsb6-python3.9-pydeck-0.7.0.drv' failed
```

Reply:
```
  checkInputs = [ pytestCheckHook pandas ];
  # tries to start a jupyter server
  disabledTests = [ "test_nbconvert" ];
```

does it for me. Think `test_nbconvert` will probably be too painful to get working inside nix's sandbox.

Reply:
done , thanks.